vuepress/docs/next/docs/component-generator/blueprint-features.md

```diff
@@ -0,0 +1,135 @@
+
+> **Note**
+>
+> custom element names (`conference-details`) require a dash in them to
+> be used, (kebab-case) - they can not be single words.
+
+For more information about web components, custom elements and micro
+frontends, refer to [Create a React micro frontend widget](../../tutorials/micro-frontends/react.html)
+
+
+### Authentication
+
+If a widget requires authentication, the component is wrapped in
+`KeycloakContext.Provider` and the Keycloak object is fetched from the
+`window.entando.keycloak` variable. Entando uses Keycloak as the
+authentication provider, but you can add any provider as needed.
+
+    ReactDOM.render(
+      <KeycloakContext.Provider value={this.keycloak}>
+        <StylesProvider jss={this.jss}>
+          <ThemeProvider theme={this.muiTheme}>{FormContainer}</ThemeProvider>
+        </StylesProvider>
+      </KeycloakContext.Provider>,
+      this.mountPoint
+    );
+
+For more information about the authentication process, please refer
+to the [Authentication section](../reference/identity-management.html#authentication).
+
+### Custom events
+
+All widgets rely on custom events for communication which is why custom
+event creation and removal is in each widget, as well as event listener
+creation. Note that when an event listener is created, it should be when
+the element is no longer needed. It should be created in the custom
+element’s `disconnectedCallback()` function.
+
+To add more events to listen to the widgets, add the event types to
+`INPUT_EVENT_TYPES` object at
+`detailsWidget/src/custom-elements/widgetEventTypes.js` which will add
+it to the listener list. To remove the event, simply remove the element from the list.
+
+For more information about custom events and widget communication,
+please refer to the section on [Widget communication](../../tutorials/micro-frontends/communication.html).
+
+### Tests and mocks
+
+Each widget has tests written for it. Entando uses
+`react-testing-library`, but developers are free to upgrade and use any
+tool desired. Tests are kept at `detailsWidget/src/components/` and mocks for
+them are at `detailsWidget/src/components/`.
+
+### PropTypes
+
+PropTypes for data used across several components are shared. You can
+see and modify them at `detailsWidget/src/components/`. This way you can
+avoid repeating the same propTypes in each component and import the shared
+ones.
+
+    import React from 'react';
+    import conferenceType from 'components/__types__/conference';
+
+    const ConferenceDetails = props => {
+      // ...
+    };
+
+    ConferenceDetails.propTypes = {
+      conference: conferenceType,
+      t: PropTypes.func.isRequired,
+    };
+
+    export default ConferenceDetails;
+
+### Fetching data
+
+For data fetching from widgets, use Fetch API. You can find the functions for
+fetching data at `detailsWidget/src/api`, in different files for
+different contexts.
+
+## Form widget
+
+For displaying forms within a widget, use
```

Comment:
To display

Reply:
I think these are helper apps. One doesn't directly do the other, Formik doesn't display the forms, it helps manage the data.